stm32-modules/heater-shaker/include/heater-shaker/motor_task.hpp

```diff
@@ -369,64 +372,83 @@ class MotorTask {
             } else {
-                cached_plate_lock_id = msg.id;
                 policy.plate_lock_set_power(OpenPower);
                 plate_lock_state.status = PlateLockState::OPENING;
                 polling_time = 0;
-                static_cast<void>(get_message_queue().try_send(
-                    messages::CheckPlateLockStatusMessage{}));
             }
-        } else {
-            plate_lock_state.status = PlateLockState::IDLE_OPEN;
-            static_cast<void>(
-                task_registry->comms->get_message_queue().try_send(
-                    messages::AcknowledgePrevious{.responding_to_id = msg.id}));
         }
+        static_cast<void>(get_message_queue().try_send(check_state_message));
     }
 
     template <typename Policy>
     auto visit_message(const messages::ClosePlateLockMessage& msg,
                        Policy& policy) -> void {
         static constexpr float ClosePower = 1.0F;
-        if ((!policy.plate_lock_closed_sensor_read()) &&
-            (plate_lock_state.status != PlateLockState::IDLE_CLOSED)) {
+        auto check_state_message = messages::CheckPlateLockStatusMessage{
+            .responding_to_id = msg.id, .from_startup = msg.from_startup};
+        if ((policy.plate_lock_closed_sensor_read()) ||
+            (plate_lock_state.status == PlateLockState::IDLE_CLOSED)) {
+            plate_lock_state.status = PlateLockState::IDLE_CLOSED;
+        } else {
             if ((state.status != State::STOPPED_HOMED) &&
                 (state.status != State::STOPPED_UNKNOWN)) {
-                static_cast<void>(
-                    task_registry->comms->get_message_queue().try_send(
-                        messages::AcknowledgePrevious{
-                            .responding_to_id = msg.id,
-                            .with_error =
-                                errors::ErrorCode::MOTOR_NOT_STOPPED}));
+                check_state_message.with_error =
+                    errors::ErrorCode::MOTOR_NOT_STOPPED;
             } else {
-                from_startup = msg.from_startup ? true : false;
-                cached_plate_lock_id = msg.id;
                 policy.plate_lock_set_power(ClosePower);
                 plate_lock_state.status = PlateLockState::CLOSING;
                 polling_time = 0;
-                static_cast<void>(get_message_queue().try_send(
-                    messages::CheckPlateLockStatusMessage{}));
             }
-        } else {
-            plate_lock_state.status = PlateLockState::IDLE_CLOSED;
+        }
+        static_cast<void>(get_message_queue().try_send(check_state_message));
+    }
+
+    template <typename Policy>
+    auto visit_message(const messages::CheckPlateLockStatusMessage& msg,
+                       Policy& policy) -> void {
+        if (msg.with_error != errors::ErrorCode::NO_ERROR) {
+            static_cast<void>(
+                task_registry->comms->get_message_queue().try_send(
```

Comment:
this probably doesn't want to happen if `msg.from_startup == true` right?

Reply:
Motor shouldn't be running if from startup so we should never hit this condition. I think this is a good protection to have, but maybe don't create an error message? But want to give the user some feedback...

Reply:
I think it's a good error message to have for the normal case. Your point is good, let's just keep it.